rust-runtime/aws-smithy-types-convert/src/instant.rs

```diff
@@ -0,0 +1,234 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+//! Conversions from [`Instant`] to the types in the
+//! [`time`](https://crates.io/crates/time) or
+//! [`chrono`](https://crates.io/crates/chrono)
+//! crates.
+
+use aws_smithy_types::instant::Instant;
+use std::error::Error as StdError;
+use std::fmt;
+
+/// Conversion error
+#[non_exhaustive]
+#[derive(Debug)]
+pub enum Error {
+    /// Conversion failed because the value being converted is out of range for its destination
+    #[non_exhaustive]
+    OutOfRange(Box<dyn StdError + Send + Sync + 'static>),
```

Comment:
is this error type actually unknown? I figured it would always be `DateTimeParseError`

Reply:
I made this one open ended so that we can add additional library conversions to the `aws-smithy-types-convert` crate without having to change the error type at all. In this case, the only error being fed into `Error::OutOfRange` is the error from `time::OffsetDateTime::from_unix_timestamp_nanos`.